lib/git/diff.rb

```diff
@@ -89,10 +89,10 @@ def binary?
       end
 
       def blob(type = :dst)
-        if type == :src
-          @base.object(@src) if @src != '0000000'
-        else
-          @base.object(@dst) if @dst != '0000000'
+        if type == :src && !/0{7,40}/.match(@src)
```

Comment:
Perhaps worth mentioning, I couldn't find a test that required the check for `@src != 0000000` to be there. Assuming there's some edge case it's needed for, I changed it to the regex version.

Reply:
seconded, there is no check for `@src`